thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/translator/DetectionConfigTranslator.java

```diff
@@ -530,6 +535,11 @@ private DetectionConfigDTO generateDetectionConfig(Map<String, Object> yamlConfi
     config.setActive(MapUtils.getBooleanValue(yamlConfigMap, PROP_ACTIVE, true));
     config.setYaml(yamlConfig);
 
+    //TODO: data-availability trigger is only enabled for detections running on PINOT only
+    if (MapUtils.getString(yamlConfigMap, PROP_CRON) == null
+        && datasetConfigs.stream().allMatch(c -> c.getDataSource().equals(PinotThirdEyeDataSource.DATA_SOURCE_NAME))) {
```

Comment:
Need to check if this is daily/weekly detection.

Reply:
Added the check for the granularity of dataset. Even if the detection is a weekly detection, it can be data-availability triggered as long as it is daily Pinot dataset.